sionna/channel/optical/fiber.py

```diff
@@ -0,0 +1,309 @@
+#
+# SPDX-FileCopyrightText: Copyright (c) 2021-2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Author: Tim Alexander Uhlemann <uhlemann@ieee.org>
+
+"""
+This module defines the split-step Fourier method to approximate the solution of
+the nonlinear Schroedinger equation.
+"""
+
+import tensorflow as tf
+from tensorflow.keras.layers import Layer
+import sionna
+from sionna.channel import utils
+
+
+class SSFM(Layer):
```

Comment:
Pylint shows the following unused arguments in some of the functions. Are they really needed?

sionna/channel/optical/fiber.py:194:29: W0613: Unused argument 'dz' (unused-argument)
sionna/channel/optical/fiber.py:248:20: W0613: Unused argument 'q' (unused-argument)
sionna/channel/optical/fiber.py:248:23: W0613: Unused argument 'precalculations' (unused-argument)

Reply:
First one was indeed unused. The latter two are due to the tensorflow requirement that the cond-method of the while-loop has to have the same signature as the corresponding body (without ever using the arguments). I added "# pylint: disable=unused-argument" in this block.